version: '3.7'
services:
  plex:
    image: linuxserver/plex:latest
    container_name: plex
    restart: unless-stopped
    mem_limit: 2G
    networks:
      - traefik_network
    environment:
      - PUID=1200
      - PGID=1200
      - TZ=Europe/Paris
      - ADVERTISE_IP=plex.chinour.fr:443
    volumes:
      - ./config/plex:/config
      - /mediastorage/films:/data/movies
      - /mediastorage/series:/data/tv
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik_network"
      - "traefik.http.routers.plex.entrypoints=web,websecure"
      - "traefik.http.routers.plex.rule=Host(`plex.pi.chinour.fr`)"
      - "traefik.http.services.plex.loadbalancer.server.port=32400"
      - "traefik.http.routers.plex.tls=true"
      - "traefik.http.routers.plex.tls.certresolver=leresolver"

  transmission:
    image: linuxserver/transmission
    container_name: transmission
    restart: unless-stopped
    mem_limit: 2G
    networks:
      - traefik_network
    environment:
      - TZ=Europe/Paris
      - PUID=1200
      - PGID=1200
    ports:
      - "51413:51413"
      - "51413:51413/udp"
    volumes:
      - ./config/transmission:/config
      - /mediastorage:/mediacenter
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik_network"
      - "traefik.http.routers.plex.entrypoints=web,websecure"
      - "traefik.http.routers.plex.rule=Host(`torrent.chinour.fr`)"
      - "traefik.http.services.plex.loadbalancer.server.port=9091"
      - "traefik.http.routers.plex.tls=true"
      - "traefik.http.routers.plex.tls.certresolver=leresolver"

networks:
  traefik_network:
    external: true
